src/Engine/GameInput/SelectActionTarget.java

```diff
@@ -23,15 +23,29 @@ protected OptionSet initOptions()
     ArrayList<XYCoord> targets = myStateData.actionSet.getTargetedLocations();
 
     consider(targets.get(0));
+
+    boolean useFreeSelect = myStateData.actionSet.useFreeSelect;
+    // For constrained select, switch to free-tile select in target-rich environments
+    if( !useFreeSelect && targets.size() > 3 )
+    {
+      int maxDist = 0;
+      for( XYCoord a : targets )
+        for( XYCoord b : targets )
+          maxDist = Math.max(maxDist, a.getDistance(b));
+
+      useFreeSelect = maxDist < targets.size();
```

Comment:
This is a bit weird. But OK.

Reply:
It means you have to be denser than a straight line for it to "force" free tile select.
Could also make this system an option, now that we have a special page for that.